core/iwasm/libraries/libc-wasi/sandboxed-system-primitives/src/blocking_op.c

```diff
@@ -8,119 +8,67 @@
 #include "ssp_config.h"
 #include "blocking_op.h"
 
-int
-blocking_op_close(wasm_exec_env_t exec_env, int fd)
+__wasi_errno_t
+blocking_op_close(wasm_exec_env_t exec_env, os_handle handle)
 {
     if (!wasm_runtime_begin_blocking_op(exec_env)) {
-        errno = EINTR;
-        return -1;
+        return __WASI_EINTR;
```

Comment:
I'm not sure I completely understand - where these blocking op (file) functions are called, we don't check `errno` after the function call since we can obtain the error directly from the return value of the function. But don't mind either way really, just don't know if it has any use setting errno here.

Reply:
I am not sure, the code was added by yamt in PR https://github.com/bytecodealliance/wasm-micro-runtime/pull/2516/files, I think it was needed by him. But now it seems good to directly return __WASI_EINTR.

Reply:
Ok I'll resolve this then if that's ok. Seems like we agree it's sufficient (and probably cleaner) to propagate the error via the return value rather than errno.